13-translations.Rmd

```diff
@@ -0,0 +1,71 @@
+# Translations
+
+This chapter covers internationalization in R, i.e., the display of messages in languages other than English.
+R leverages the [`gettext`](https://www.gnu.org/software/gettext/) program to handle the conversion from English
+to arbitrary target languages.
+
+Having messages available in other languages can be an important bridge for R learners not confident in English --
+rather than learning two things at once (coding in R and processing diagnostic information in English), they can
+focus on coding while getting more natural errors/warnings in their native tongue.
+
+The [`gettext` manual](https://www.gnu.org/software/gettext/manual/index.html) is a more canonical reference for
+deep understanding of how `gettext` works. This chapter will just give a broad overview of how it works, with
+particular focus on how things work for R, with the goal of making it as low-friction as possible for developers
+and users to contribute new/updated translations.
+
+## How translations work
+
+Each of the default packages distributed with R (i.e., those found in `./src/library` such as `base`, `utils`,
```

Comment:
```suggestion
There are two basic files with extensions `.pot` and `.po` that are usually required during the Translation process in R. The `.pot` files are the template files which contain the error messages, warnings, and other similar messages, in R. In the template `.pot` file these messages will be available in standard English against the placeholder `msgid "Standard English message is placed here"`. Below every `msgid` there will be a placeholder for the translated message called `msgstr ""` and it would always be empty (default) in the `.pot` file. The `msgstr` is to be filled in the corresponding `.po` file - to include the appropriate translation. Both the template `.pot` file and the translated `.po` file should stored in the same directory always. Each of the default packages distributed with R (i.e., those found in `./src/library` such as `base`, `utils`,
```

Reply:
Could you add a link to an example `.pot` and `.po` file from the SVN repo? May help curious readers quickly see what these files look like and what to expect.

Reply:
implementation question -- what sort of link would be appropriate? Is a relative link possible? Or should we be perma-linking a commit from the `r-devel/r-svn` repo?